### PR TITLE
build: Drop support for config-time componets

### PR DESCRIPTION
**Pull request by bwbarrett**
Remove support for both dropping in components to a tarball (without re-running autogen) and for using full configure scripts (which, practically speaking, was required for the first capability).  We had accidentally broken both a couple months ago, and don't have any tests for it, so just remove.

### CODE REVIEW

File: config/opal_mca.m4 [hunk 1]
```diff
@@ -859,13 +786,12 @@ AC_MSG_ERROR([*** $2 component $3 was supposed to be direct-called, but
     # Since a configure script component can never be used in a
     # STOP_AT_FIRST framework, we don't have to implement the else
     # clause in the literal check.
-    AS_LITERAL_IF([$3],
-        [AS_IF([test "$$2_$3_WRAPPER_EXTRA_CPPFLAGS" != ""],
+     AS_IF([test "$$2_$3_WRAPPER_EXTRA_CPPFLAGS" != ""],
```
Comment (jsquyres) [inline config/opal_mca.m4:789]: Trivial: incorrect indent here.

File: config/opal_mca.m4 [hunk 2]
```diff
@@ -635,86 +634,16 @@ AC_DEFUN([MCA_CONFIGURE_M4_CONFIG_COMPONENT],[
...
-# USAGE:
-#   MCA_CONFIGURE_ALL_CONFIG_COMPONENTS(project_name,
-#                         framework_name,
-#                         all_components_variable,
-#                         static_components_variable,
-#                         dso_components_variable,
-#                         static_ltlibs_variable)
-#
-######################################################################
-AC_DEFUN([MCA_CONFIGURE_ALL_CONFIG_COMPONENTS],[
```
Comment (jsquyres) [inline config/opal_mca.m4:655]: Don't you need to also remove the invocation of this macro on line 459?

**Review by bosilca — APPROVED** [review]

File: config/opal_mca.m4 (hunk 2 above)
Comment (bwbarrett) [reply config/opal_mca.m4:655]: yup; fixed.

File: config/opal_mca.m4 (hunk 1 above)
Comment (bwbarrett) [reply config/opal_mca.m4]: Fixed; emacs failed me there.

**Review by jsquyres — APPROVED** [review]

**Comment by bwbarrett** [comment]
Last update was to fix a spelling error in the commit title.  No other changes.  Intend to commit when CI completes.

**Pull request merged**